docs/development/setup-advanced.md

```diff
@@ -136,6 +137,127 @@ Store.
+
+**NOTE**: Hyper-V is available only on Windows Enterprise, Pro, or Education.
+
+1. Get started by cloning the repository to your local machine, and linking the
+   upstream to your fork.
+
+   ```bash
+   git clone --config pull.rebase git@github.com:YOURUSERNAME/zulip.git
+   cd zulip/
+   git remote add -f upstream https://github.com/zulip/zulip.git
+   ```
+
+2. You will have to open up powershell with administrator right. This is not a
+   limitation of Vagrant, but of Hyper-V itself. Then create the development
+   environment setup.
+
+   ```bash
+   vagrant up --provider=hyperv
+   ```
+
+   You should get output like this:
+
+   ```text
+   Bringing machine 'default' up with 'hyperv' provider...
+   ==> default: Verifying Hyper-V is enabled...
+   ==> default: Verifying Hyper-V is accessible...
+   <other stuff>...
+   ==> default: Waiting for the machine to report its IP address...
+       default: Timeout: 120 seconds
+       default: IP: 172.28.119.70
+   ==> default: Waiting for machine to boot. This may take a few minutes...
+       default: SSH address: 172.28.122.156
+   ==> default: Machine booted and ready!
+   ==> default: Preparing SMB shared folders...
+   Vagrant requires administrator access for pruning SMB shares and
+   may request access to complete removal of stale shares.
+   ==> default: Starting the machine...
+   <other stuff>...
+    default: Username (user[@domain]): <your-machine-username>
+    default: Password (will be hidden):
+   ```
+
+   At this point, you will be prompted for your administrator username and password.
+   This should be your administrator account and password credentials (and not your
+   Microsoft account credentials).
+
+3. SSH into your newly created virtual machine
+
+   ```bash
+   vagrant ssh
+   ```
+
+   This will drop you into the bash shell of the Zulip development environment
+   where you can execute bash commands.
+
+4. Set the `EXTERNAL_HOST` environment variable.
+
+   ```bash
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ export EXTERNAL_HOST="$(hostname -I | xargs):9991"
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ echo $EXTERNAL_HOST
+   ```
+
+   The output will be like:
+
+   ```text
+   172.28.122.156:9991
+   ```
+
+   Make sure you note down this down. This is where your zulip development web
+   server can be accessed.
+
+   ```eval_rst
+   .. important::
+      The output of the above command changes every time you restart the Vagrant
+      development machine. Thus, it will have to be run every time you bring one up.
+      The Zulip developers are currently in the process of deciding the best method
+      to automate this process.
+   ```
+
+5. If all went well, you should now be able to bring up the Zulip development web server.
+
+   ```bash
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ ./tools/run-dev.py
+   ```
+
+   The output will look like:
+
+   ```text
+   Starting Zulip on http://172.28.122.156:9991:9991/.  Internal ports:
```

Comment:
The port is appearing twice. Also, the output has changed recently, so you should probably update this,